parquet/src/main/java/org/apache/iceberg/parquet/ParquetReader.java

```diff
@@ -79,9 +83,11 @@ private ReadConf<T> init() {
               nameMapping,
               reuseContainers,
               caseSensitive,
-              null);
-      this.conf = readConf.copy();
-      return readConf;
+              null)) {
+        this.conf = readConf.copy();
+      } catch (IOException e) {
+        LOG.warn("Failed to close ReadConf", e);
```

Comment:
I think it would make more sense to catch+log this inside `ReadConf.close()` so that `ReadConf.close()` doesn't throw anything